lib/index.js

```diff
@@ -319,3 +355,114 @@ exports.socket = createSocket;
 
 exports.createDevice = createDevice;
 exports.device = createDevice;
+
+/* convenience methods to set and to get sockopts, a.k.a. the sockopt API */
+
+function linger(number){
+  if(number){
+    if(setopt(this.binding, 'linger', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' linger@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'linger');
+  }
+}
+function sndbuf(number){
+  if(number){
+    if(setopt(this.binding, 'sndbuf', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' sndbuf@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'sndbuf');
+  }
+}
+
+function rcvbuf(number){
+  if(number){
+    if(setopt(this.binding, 'rcvbuf', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type +' rcvbuf@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'rcvbuf');
+  }
+}
+
+function sndtimeo(number){
+  if(number){
+    if(setopt(this.binding, 'sndtimeo', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'sndtimeo');
+  }
+}
+
+function rcvtimeo(number){
+  if(number){
+    if(setopt(this.binding, 'rcvtimeo', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'rcvtimeo');
+  }
+}
+
+function reconn(number){
+  if(number){
+    if(setopt(this.binding, 'reconn', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' reconn@' + number + '\n');
+  } else {
+    return getopt(this.binding, 'reconn');
+  }
+}
+
+function maxreconn(number){
+  if(number){
+    if(setopt(this.binding, 'maxreconn', number) > -1) return true;
+    throw new Error(nn.Err() + ': ' + this.type + ' maxreconn@' + number +'\n');
+  } else {
+    return getopt(this.binding, 'maxreconn');
+  }
+}
```

Comment:
If someone wants to set maxreconn to `0`, this is going to act like a getter, not a setter.  Same for the others that expect a number.